driver-core/src/main/com/mongodb/internal/connection/MultiServerCluster.java

```diff
@@ -76,7 +77,9 @@ private ServerTuple(final ClusterableServer server, final ServerDescription desc
         }
     }
 
-    public MultiServerCluster(final ClusterId clusterId, final ClusterSettings settings, final ClusterableServerFactory serverFactory) {
+    // TODO: make sure that MultiServerCluster is always used for SRV hosts
```

Comment:
Whats the plan here? Would it be simpler to have a `DNSServerCluster` class for the dns lookup use case? So we dont have the dual concerns / branching logic?

Reply:
Great idea.  Done.